react-app/src/components/PortfolioScreen/PortfolioScreenAPI.ts

```diff
@@ -0,0 +1,81 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { useCosmos } from "../../api/cosmosAPI";
+import { useQueryClient } from "../../providers/QueryClientProvider";
+import { ConnectionStatus, useWallet } from "../../providers/WalletProvider";
+import { translateAddress } from "../../utils/address";
+import { useStaking } from "../../api/stakingAPI";
+import {
+  RequestState,
+  RequestStateError,
+  RequestStateInitial,
+  RequestStateLoaded,
+  RequestStateLoading,
+} from "../../models/RequestState";
+import { Portfolio } from "./PortfolioScreenModel";
+
+type PortfolioRequestState = RequestState<Portfolio | null>;
+
+export function usePortfolioQuery(): PortfolioRequestState {
+  const [loading, setLoading] = useState(false);
+  const [data, setData] = useState<Portfolio>();
+  const [error, setError] = useState<Error>();
+
+  const wallet = useWallet();
+  const cosmosAPI = useCosmos();
+  const staking = useStaking();
```

Comment:
I am thinking if it's necessary to put non-gql queries in an API file 🤔 @chihimng any thoughts?

Reply:
probably can handle the state using `const [requestState, setRequestState] = useState<RequestState>()`?